typing/ikinds/ikind.mli

```diff
@@ -0,0 +1,92 @@
+(*-------------------------------------------------------------------------*)
+(* Types shared by ikind algorithms                                        *)
+(*-------------------------------------------------------------------------*)
+
+module RigidName : sig
+  (* Rigid names are the variables that may occur in ikind formulas. *)
+
+  (* CR jujacobs: see if we can make this type abstract. *)
```

Comment:
I think it's fine that this isn't abstract. But I do think it'd be useful to use an abstract type of the id of `Unknown` names.